test_noobaa_sts_host_node_failure[noobaa-db-pg-False], test_noobaa_sts_host_node_failure[noobaa-db-pg-True] - a fix is needed

Relevant runs: 
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/41375/1916308/1916312/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/852/41375/1916308/1916310/log

Backtrace: 
elf = <mcg.test_host_node_failure.TestNoobaaSTSHostNodeFailure object at 0x7fc3629bb550>
noobaa_sts = 'noobaa-db-pg', respin_noobaa_operator = True
mcg_obj = <ocs_ci.ocs.resources.mcg.MCG object at 0x7fc36c59d7d0>
bucket_factory = <function bucket_factory_fixture.._create_buckets at 0x7fc359d41a80>
nodes = <ocs_ci.ocs.platform_nodes.IBMCloud object at 0x7fc362076dd0>
node_restart_teardown = None


@pytest.mark.parametrize(
    argnames=["noobaa_sts", "respin_noobaa_operator"],
    argvalues=[
        pytest.param(
            *[constants.NOOBAA_CORE_STATEFULSET, False],
            marks=pytest.mark.polarion_id("OCS-2672"),
        ),
        pytest.param(
            *[constants.NOOBAA_DB_STATEFULSET, False],
            marks=pytest.mark.polarion_id("OCS-2668"),
        ),
        pytest.param(
            *[constants.NOOBAA_CORE_STATEFULSET, True],
            marks=pytest.mark.polarion_id("OCS-2669"),
        ),
        pytest.param(
            *[constants.NOOBAA_DB_STATEFULSET, True],
            marks=pytest.mark.polarion_id("OCS-2670"),
        ),
    ],
)
def test_noobaa_sts_host_node_failure(
    self,
    noobaa_sts,
    respin_noobaa_operator,
    mcg_obj,
    bucket_factory,
    nodes,
    node_restart_teardown,
):
    """
    Test case to fail node where NooBaa Statefulset pod (noobaa-core, noobaa-db)
    is hosted and verify the pod is rescheduled on a healthy node

    """
    executor = ThreadPoolExecutor(max_workers=1)
    pod_obj = OCP(
        kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"]
    )

    # Get noobaa statefulset pod and node where it is hosted

  noobaa_sts_pod = get_noobaa_pods(noobaa_label=self.labels_map[noobaa_sts])[0]
E       IndexError: list index out of range



tests/functional/object/mcg/test_host_node_failure.py:104: IndexError

--- a/tests/functional/object/mcg/test_host_node_failure.py
+++ b/tests/functional/object/mcg/test_host_node_failure.py
@@ -48,7 +48,10 @@ class TestNoobaaSTSHostNodeFailure(ManageTest):
     def labels_map(self):
         ocs_version = version.get_semantic_ocs_version_from_config()
         if ocs_version >= version.VERSION_4_19:
-            noobaa_db_label = constants.NOOBAA_DB_LABEL_419_AND_ABOVE
+            noobaa_db_label = (
+                f"{constants.NOOBAA_DB_LABEL_419_AND_ABOVE},"
+                f"{constants.NB_DB_PRIMARY_POD_LABEL}"
+            )
         else:
             noobaa_db_label = constants.NOOBAA_DB_LABEL_47_AND_ABOVE
         return {

--- a/tests/functional/object/mcg/test_host_node_failure.py
+++ b/tests/functional/object/mcg/test_host_node_failure.py
@@ -40,7 +40,7 @@ log = logging.getLogger(__name__)
 @skipif_ocs_version("<4.9")
 class TestNoobaaSTSHostNodeFailure(ManageTest):
     """
-    Test to verify NooBaa Statefulset pods recovers in case of a node failure
+    Test to verify NooBaa Statefulset pods recover in case of a node failure
 
     """
 

--- a/tests/functional/object/mcg/test_host_node_failure.py
+++ b/tests/functional/object/mcg/test_host_node_failure.py
@@ -19,6 +19,7 @@ from ocs_ci.helpers.helpers import (
 from ocs_ci.ocs import constants, node
 from ocs_ci.ocs.exceptions import TimeoutExpiredError, CommandFailed
 from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import version
 from ocs_ci.ocs.resources.pod import (
     get_pod_node,
     get_noobaa_pods,
@@ -43,11 +44,18 @@ class TestNoobaaSTSHostNodeFailure(ManageTest):
 
     """
 
-    labels_map = {
-        constants.NOOBAA_CORE_STATEFULSET: constants.NOOBAA_CORE_POD_LABEL,
-        constants.NOOBAA_DB_STATEFULSET: constants.NOOBAA_DB_LABEL_47_AND_ABOVE,
-        constants.NOOBAA_OPERATOR_DEPLOYMENT: constants.NOOBAA_OPERATOR_POD_LABEL,
-    }
+    @property
+    def labels_map(self):
+        ocs_version = version.get_semantic_ocs_version_from_config()
+        if ocs_version >= version.VERSION_4_19:
+            noobaa_db_label = constants.NOOBAA_DB_LABEL_419_AND_ABOVE
+        else:
+            noobaa_db_label = constants.NOOBAA_DB_LABEL_47_AND_ABOVE
+        return {
+            constants.NOOBAA_CORE_STATEFULSET: constants.NOOBAA_CORE_POD_LABEL,
+            constants.NOOBAA_DB_STATEFULSET: noobaa_db_label,
+            constants.NOOBAA_OPERATOR_DEPLOYMENT: constants.NOOBAA_OPERATOR_POD_LABEL,
+        }
 
     @pytest.fixture(autouse=True)
     def init_sanity(self):
@@ -101,15 +109,24 @@ class TestNoobaaSTSHostNodeFailure(ManageTest):
         )
 
         # Get noobaa statefulset pod and node where it is hosted
-        noobaa_sts_pod = get_noobaa_pods(noobaa_label=self.labels_map[noobaa_sts])[0]
+        noobaa_sts_pods = get_noobaa_pods(noobaa_label=self.labels_map[noobaa_sts])
+        assert (
+            noobaa_sts_pods
+        ), f"No pods found with label {self.labels_map[noobaa_sts]}"
+        noobaa_sts_pod = noobaa_sts_pods[0]
         noobaa_sts_pod_node = get_pod_node(noobaa_sts_pod)
         log.info(f"{noobaa_sts_pod.name} is running on {noobaa_sts_pod_node.name}")
 
         # Get the NooBaa operator pod and node where it is hosted
         # Check if NooBaa operator and statefulset pod are hosted on same node
-        noobaa_operator_pod = get_noobaa_pods(
+        noobaa_operator_pods = get_noobaa_pods(
             noobaa_label=self.labels_map[constants.NOOBAA_OPERATOR_DEPLOYMENT]
-        )[0]
+        )
+        assert noobaa_operator_pods, (
+            f"No pods found with label "
+            f"{self.labels_map[constants.NOOBAA_OPERATOR_DEPLOYMENT]}"
+        )
+        noobaa_operator_pod = noobaa_operator_pods[0]
         noobaa_operator_pod_node = get_pod_node(noobaa_operator_pod)
         log.info(
             f"{noobaa_operator_pod.name} is running on {noobaa_operator_pod_node.name}"

--- a/tests/functional/object/mcg/test_host_node_failure.py
+++ b/tests/functional/object/mcg/test_host_node_failure.py
@@ -102,8 +102,8 @@ class TestNoobaaSTSHostNodeFailure(ManageTest):
         node_restart_teardown,
     ):
         """
-        Test case to fail node where NooBaa Statefulset pod (noobaa-core, noobaa-db)
-        is hosted and verify the pod is rescheduled on a healthy node
+        Test case to fail the node where a NooBaa Statefulset pod (noobaa-core, noobaa-db)
+        is hosted and verify that the pod is rescheduled on a healthy node
 
         """
         executor = ThreadPoolExecutor(max_workers=1)

